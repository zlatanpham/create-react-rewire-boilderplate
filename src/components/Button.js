import React from 'react';
import styled from 'styled-components';
import tw from 'twin.macro';

const StyledButton = styled.button`
  ${tw`bg-black text-white px-10 py-4 rounded-full hover:shadow-lg hover:bg-purple-600 hover:text-red-300 transition-all duration-200 hover:scale-110 transform`}
  font-size: 20px;
  font-weight: bold;
  border: none;
`;

const Button = ({ children, ...rest }) => (
  <StyledButton {...rest}>{children}</StyledButton>
);

export default Button;
