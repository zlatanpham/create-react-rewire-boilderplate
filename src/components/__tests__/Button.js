import React from 'react';
import { render } from '@testing-library/react';
import Button from '../Button.js';

describe('<Button/>', () => {
  const { container, rerender } = render(<Button />);

  it('Should match snapshot', () => {
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Should allow to pass custom CSS classes', () => {
    rerender(<Button className="addition classes here" />);
    expect(container.firstChild.getAttribute('class')).toContain(
      'addition classes here',
    );
  });
});
