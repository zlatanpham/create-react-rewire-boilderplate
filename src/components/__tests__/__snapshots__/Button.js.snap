// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> Should match snapshot 1`] = `
<button
  class="Button__StyledButton-sc-9ohxp-0 egjzQM"
/>
`;
